File: niro/src/main/scala/nl/uva/se/sc/niro/qls/model/gui/style/Style.scala
package nl.uva.se.sc.niro.qls.model.gui.style

abstract class Style

object Style {

  def apply(widgetType: Option[nl.uva.se.sc.niro.qls.model.ast.style.WidgetType]): WidgetType = widgetType.map(Style(_)).getOrElse(DefaultWidget())

  def apply(widgetType: nl.uva.se.sc.niro.qls.model.ast.style.WidgetType): WidgetType = widgetType match {
    case nl.uva.se.sc.niro.qls.model.ast.style.SpinBox(minimum, maximum, stepSize) => SpinBox(minimum, maximum, stepSize)
    case nl.uva.se.sc.niro.qls.model.ast.style.Slider(minimum, maximum)            => Slider(minimum, maximum)
    case nl.uva.se.sc.niro.qls.model.ast.style.ComboBox(trueValue, falseValue)     => ComboBox(trueValue, falseValue)
    case nl.uva.se.sc.niro.qls.model.ast.style.Radio(trueValue, falseValue)        => Radio(trueValue, falseValue)
    case nl.uva.se.sc.niro.qls.model.ast.style.CheckBox()                          => DefaultWidget()
  }

  def apply(font: nl.uva.se.sc.niro.qls.model.ast.style.FontType): FontType = FontType(font.name)

  def apply(fontSize: nl.uva.se.sc.niro.qls.model.ast.style.FontSize) = FontSize(fontSize.size)

  def apply(color: nl.uva.se.sc.niro.qls.model.ast.style.Color) = Color(color.color)

  def apply(width: nl.uva.se.sc.niro.qls.model.ast.style.Width) = Width(width.width)

}

File: YouDontMakeFriendsWithDotNet/NoRightAnswersToWrongQuestions/QlsTransformer/Domain/Output/Tools/Style.cs
using QlsTransformer.Domain.Ast.Nodes;

namespace QlsTransformer.Domain.Output.Tools
{
    public class Style
    {
        public Style(
            Style style,
            IWidget widget = null,
            int? width = null,
            string font = null,
            decimal? fontSize = null,
            string color = null)
        {
            Widget = widget ?? style.Widget;
            Width = width ?? style.Width;
            Font = font ?? style.Font;
            FontSize = fontSize ?? style.FontSize;
            Color = color ?? style.Color;
        }

        public Style(
            IWidget widget,
            int width,
            string font,
            decimal fontSize,
            string color)
        {
            Widget = widget;
            Width = width;
            Font = font;
            FontSize = fontSize;
            Color = color;
        }

        public IWidget Widget { get; }
        public int Width { get; }
        public string Font { get; }
        public decimal FontSize { get; }
        public string Color { get; }
    }
}